Sources/NIO/SelectorGeneric.swift

```diff
@@ -383,3 +385,37 @@ enum SelectorStrategy {
     /// Try to select all ready IO at this point in time without blocking at all.
     case now
 }
+
+/// A Registration on a `Selector`, which is interested in an `SelectorEventSet`.
+/// `registrationID` is used by the event notification backends (kqueue, epoll, ...)
+/// to mark events to allow for filtering of received return values to not be delivered to a
+/// new `Registration` instance that receives the same file descriptor. Ok if it wraps.
+/// Needed for i.e. testWeDoNotDeliverEventsForPreviouslyClosedChannels to succeed.
+@usableFromInline struct SelectorRegistrationID: Hashable {
+    @usableFromInline var _rawValue: UInt32
+
+    @inlinable var rawValue: UInt32 {
+        return self._rawValue
+    }
+
+    @inlinable static var initialRegistrationID: SelectorRegistrationID {
+        return SelectorRegistrationID(rawValue: .max)
```

Comment:
@Lukasa I changed this to use `UInt32.max` so we're sure we run the overflow codepath always.

Reply:
CC @hassila too, just in case you get reg ids that look weird :)